### PR TITLE
(1) Add some JudgeCases (10017/70038) (2) ProvidedCodes's Nullity in AllMatchPlugin (4) Fix ExamHomePresenter NPE (5) Upload TestcaseIO TIMEOUT fix

### PR DESCRIPTION
**Pull request by Johnny850807**
(1) Add some JudgeCases (10017/70038) (2) ProvidedCodes's Nullity in AllMatchPlugin (4) Fix ExamHomePresenter NPE (5) Upload TestcaseIO TIMEOUT fix

### CODE REVIEW

File: Judger/src/main/java/tw/waterball/judgegirl/judger/CCJudger.java [hunk 1]
```diff
@@ -143,11 +143,13 @@ protected void downloadSubmittedCodes() throws IOException {
...
     protected void downloadProvidedCodes() throws IOException {
         LanguageEnv languageEnv = getLanguageEnv();
-        try (FileResource zip = problemServiceDriver.downloadProvidedCodes(
-                getProblem().getId(), languageEnv.getName(), languageEnv.getProvidedCodesFileId())) {
-            ZipUtils.unzipToDestination(zip.getInputStream(), getSourceRootPath());
-        } catch (NotFoundException ignored) {
-            logger.info("No providedCodes.");
+        if (languageEnv.getProvidedCodesFileId() != null) {
+            try (FileResource zip = problemServiceDriver.downloadProvidedCodes(
+                    getProblem().getId(), languageEnv.getName(), languageEnv.getProvidedCodesFileId())) {
```
Comment (Wally5077) [inline Judger/src/main/java/tw/waterball/judgegirl/judger/CCJudger.java:148]: Please declare a variable 'providedCodesFileId' for ```languageEnv.getProvidedCodesFileId();```

**Review by Wally5077 — CHANGES_REQUESTED** [review]
review done

File: Judger/src/main/java/tw/waterball/judgegirl/judger/CCJudger.java (hunk 1 above)
Comment (Johnny850807) [reply Judger/src/main/java/tw/waterball/judgegirl/judger/CCJudger.java:148]: Thank you but I don't think this helps.

Comment (Wally5077) [reply Judger/src/main/java/tw/waterball/judgegirl/judger/CCJudger.java:148]: Ok. You can merge this pr.

**Pull request merged**